### PR TITLE
Add RBAC and PSP

### PR DESCRIPTION
**Pull request by StianOvrevage**
This tries to update and add the necessary things to install the VM chart on clusters with RBAC and PodSecurityPolicy enabled.

I have tested this on K8s v1.14.8 running on Azure AKS with PSP enabled.

I have taken most of the config from the `stable/grafana` chart.

Here is a default install with RBAC and PSP enabled:

```
helm3 upgrade --install vm0 charts/victoria-metrics-cluster

$ kubectl get pods -w
NAME                                                     READY   STATUS    RESTARTS   AGE
vm0-victoria-metrics-cluster-vminsert-6646dc68c5-bqr2m   1/1     Running   0          9m57s
vm0-victoria-metrics-cluster-vminsert-6646dc68c5-jppmn   1/1     Running   0          9m57s
vm0-victoria-metrics-cluster-vmselect-67c5988fd9-jmwqf   1/1     Running   0          9m57s
vm0-victoria-metrics-cluster-vmselect-67c5988fd9-z5bk9   1/1     Running   0          9m57s
vm0-victoria-metrics-cluster-vmstorage-0                 1/1     Running   0          9m57s
vm0-victoria-metrics-cluster-vmstorage-1                 1/1     Running   0          8m58s

$ kubectl describe pod vm1-victoria-metrics-cluster-vminsert-798645c97d-8r4k7
Annotations:        kubernetes.io/psp: vm1-victoria-metrics-cluster
```

And namespaced:
```
$ kubectl create ns vm
$ helm3 upgrade --namespace vm --install vm1 charts/victoria-metrics-cluster --set rbac.namespaced=true

$ kubectl get pods -n vm -w
NAME                                                    READY   STATUS    RESTARTS   AGE
vm1-victoria-metrics-cluster-vminsert-b8fbb6858-6mslb   1/1     Running   0          23s
vm1-victoria-metrics-cluster-vminsert-b8fbb6858-fcz7g   1/1     Running   0          23s
vm1-victoria-metrics-cluster-vmselect-96df864b8-dkcl6   1/1     Running   0          23s
vm1-victoria-metrics-cluster-vmselect-96df864b8-r9n85   1/1     Running   0          23s
vm1-victoria-metrics-cluster-vmstorage-0                1/1     Running   0          23s
vm1-victoria-metrics-cluster-vmstorage-1                1/1     Running   0          0s
```

Right now all components use the same config. We could probably add more fine-grained control but I think this should be a good place to start iterating from at least :)

### CODE REVIEW

**Comment by tenmozes** [comment]
@StianOvrevage thank you for contribution, will take a look soon

File: charts/victoria-metrics-cluster/templates/_helpers.tpl [hunk 1]
```diff
@@ -6,13 +6,42 @@ Expand the name of the chart.
 {{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 
+{{/*
```
Comment (tenmozes) [inline charts/victoria-metrics-cluster/templates/_helpers.tpl]: please bump version of Chart in `Chart.yaml`

File: charts/victoria-metrics-cluster/values.yaml [hunk 2]
```diff
@@ -7,6 +7,17 @@
 ##
 clusterDomainSuffix: cluster.local
 
+rbac:
+  create: true
+  pspEnabled: true
+  namespaced: false
```
Comment (tenmozes) [inline charts/victoria-metrics-cluster/values.yaml]: is it better to remove this param and always create ClusterRoleBinding?

Comment (StianOvrevage) [reply charts/victoria-metrics-cluster/values.yaml]: I don't think so. For us for example we would like to have it contained to a namespace. It should also make it possible for "regular" users to install the chart. While creating ClusterRole(Binding)s require ClusterAdmin privileges I think.

File: charts/victoria-metrics-cluster/templates/_helpers.tpl (hunk 1 above)
Comment (StianOvrevage) [reply charts/victoria-metrics-cluster/templates/_helpers.tpl]: Ofc! My bad!

File: charts/victoria-metrics-cluster/templates/clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{{- if and .Values.rbac.create (not .Values.rbac.namespaced) }}
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  labels:
+    app: {{ template "victoria-metrics.name" . }}
```
[the remark this answers is not in the document]
Comment (StianOvrevage) [reply charts/victoria-metrics-cluster/templates/clusterrole.yaml]: Done. There was some variation in how it was done already but I've tried to make it as similar as possible.

**Comment by StianOvrevage** [comment]
Have a look now :)

File: charts/victoria-metrics-cluster/Chart.yaml [hunk 4]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 appVersion: 1.29.3
 description: Victoria Metrics Cluster version - high-performance, cost-effective and scalable TSDB, long-term remote storage for Prometheus
 name: victoria-metrics-cluster
-version: 0.1.0
+version: 0.1.1
```
Comment (tenmozes) [inline charts/victoria-metrics-cluster/Chart.yaml]: the latest version is 0.2.0 could please bump to 0.3.0

File: charts/victoria-metrics-cluster/templates/podsecuritypolicy.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+metadata:
+  name: {{ template "victoria-metrics.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+  {{- include "victoria-metrics.common.metaLabels" . | nindent 4 }}
+  {{- if .Values.rbac.extraLabels }}
+  {{ toYaml .Values.rbac.extraLabels | indent 4}}
+  {{- end }}
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: 'docker/default'
```
Comment (tenmozes) [inline charts/victoria-metrics-cluster/templates/podsecuritypolicy.yaml]: does user need to add more annotation here which can be specified in values.yaml?

Comment (StianOvrevage) [reply charts/victoria-metrics-cluster/templates/podsecuritypolicy.yaml]: Not sure. We do not but someone might. I have added it now.

File: charts/victoria-metrics-cluster/Chart.yaml (hunk 4 above)
Comment (StianOvrevage) [reply charts/victoria-metrics-cluster/Chart.yaml]: Done

**Comment by StianOvrevage** [comment]
And thank you! 😄

**Comment by tenmozes** [comment]
i will resolve conflict and merge tomorrow

**Pull request merged**